cvat-canvas/src/typescript/canvasView.ts

```diff
@@ -3116,6 +3120,27 @@ export class CanvasViewImpl implements CanvasView, Listener {
                 `${withSource ? `(${source})` : ''}`).style({
                     'text-transform': 'uppercase',
                 });
+                if (withDimensions && ['rectangle', 'ellipse'].includes(state.shapeType)) {
+                    let width = state.points[2] - state.points[0];
+                    let height = state.points[3] - state.points[1];
+
+                    if (state.shapeType === 'ellipse') {
+                        width *= 2;
+                        height *= -2;
+                    }
+
+                    const normWidth = width.toFixed(1);
+                    const normHeight = height.toFixed(1);
+                    const normRot = state.rotation === 0 ? '' : `${state.rotation.toFixed(1)}\u00B0`;
+
+                    block
+                        .tspan(`${normWidth}x${normHeight}px ${normRot}`)
```

Comment:
We have similar code in [shared.ts](https://github.com/cvat-ai/cvat/blob/develop/cvat-canvas/src/typescript/shared.ts#L110). Maybe there is a way to refactor this logic into separate funciton so we dont do the same thing in 2 places?

Reply:
At least generation of text